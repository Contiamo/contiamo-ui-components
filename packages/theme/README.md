# Operational UI Theme

The `@operational/theme` package is basically a structured plain JavaScript object that other `@operational` packages consume to customize their styling.

## Getting Started

Install the package via npm or yarn:

`npm install @operational/theme @operational/components`

Inspect and customize the theme object:

```js
import { operationalTheme } from "@operational/theme"

const ownTheme = {
  ...operationalTheme,
  colors: {
    ...operationalTheme.colors,
    success: "#0F0"
  }
}
```

Use this theme with e.g. `@operational/components`, like so:

```js
import React from "react"
import { OperationalUI, Button, ThemeProvider } from "@operational/components"

const App = () => {
  <OperationalUI, ThemeProvider theme={ownTheme}>
    <Button color="success">So much success</Button>
  </OperationalUI, ThemeProvider>
}
```

The background color of the button will now turn to what you specified.

## Documentation

Head to the theme section of the [documentation site](https://ui.contiamo.com/documentation/theming) to see an overview of what you can do with the theme.

