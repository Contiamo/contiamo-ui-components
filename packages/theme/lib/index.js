"use strict";
// Type definitions
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
// Default theme definition
var colors = {
    brand: "#000000",
    info: "#1499CE",
    success: "#00b34d",
    warning: "#FFAE00",
    error: "#DE1A1A",
    white: "#FFFFFF",
    black: "#000000",
    gray10: "#F8F8F8",
    gray20: "#e8e8e8",
    gray30: "#D0D0D0",
    gray40: "#C6C6C6",
    gray50: "#BBBBBB",
    gray60: "#999999",
    gray70: "#808080",
    gray80: "#747474",
    gray90: "#444444",
    background: "#F1F1F1",
    bodyText: "#555f61",
    cardBackground: "#FFFFFF",
    border: "#dadada",
    emphasizedText: "#373d3f",
    lightText: "#969696",
    linkText: "#1499CE",
    sidenavBackground: "#393939",
    separator: "#f2f2f2",
    secondarySeparator: "#f8f8f8"
};
var baseTypography = {
    lineHeight: "1.5",
    textTransform: "none",
    letterSpacing: "normal"
};
var typography = {
    title: __assign({}, baseTypography, { fontSize: 22, fontWeight: 600 }),
    heading1: __assign({}, baseTypography, { fontSize: 13, fontWeight: 700, color: "#373d3f" }),
    heading2: __assign({}, baseTypography, { fontSize: 13, fontWeight: 600, textTransform: "uppercase", color: "#969696", "&::before": {
            content: "Â» "
        } }),
    body: __assign({}, baseTypography, { fontSize: 13, fontWeight: 400 }),
    small: __assign({}, baseTypography, { fontSize: 12, fontWeight: 400 })
};
var shadows = {
    pressed: "inset 0 1px 1px rgba(0,0,0,0.15)",
    card: "0px 1px 2px #d3d1d1",
    focus: "inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(82,168,236,.6)",
    popup: "0 3px 12px rgba(0, 0, 0, .14)"
};
var operational = {
    typography: typography,
    shadows: shadows,
    colors: colors,
    spacing: 12,
    fontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
    baseZIndex: 0
};
exports.operational = operational;
//# sourceMappingURL=index.js.map