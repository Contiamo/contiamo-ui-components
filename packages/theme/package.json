{
  "name": "@operational/theme",
  "version": "0.1.0-16",
  "description": "Customization theme for the Operational UI packages.",
  "main": "./lib/index.js",
  "types": "./lib/index.d.ts",
  "repository": "git@github.com:Contiamo/operational-ui.git",
  "author": "Peter Szerzo <peter@contiamo.com>",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "package": "rm -rf lib && tsc -d",
    "package:watch": "tsc -d -w",
    "lint": "tslint './src/**/*.{ts,tsx}' --fix",
    "prettier": "prettier './src/**/*.{ts,tsx}' --write",
    "test": "jest",
    "test:u": "jest -u"
  },
  "devDependencies": {
    "@types/core-js": "0.9.43",
    "@types/enzyme": "3.1.5",
    "@types/jest": "20.0.8",
    "@types/lodash": "4.14.74",
    "@types/marked": "0.3.0",
    "@types/node": "8.0.27",
    "@types/tinycolor2": "1.4.0",
    "awesome-typescript-loader": "3.4.1",
    "enzyme": "2.9.1",
    "enzyme-to-json": "1.5.1",
    "jest": "21.2.1",
    "jest-enzyme": "3.8.1",
    "jest-glamor-react": "3.1.0",
    "jest-serializer-enzyme": "1.0.0",
    "prettier": "1.8.2",
    "react-test-renderer": "16.0.0",
    "ts-jest": "21.0.0",
    "tslint": "5.7.0",
    "tslint-config-airbnb": "5.2.1",
    "tslint-config-prettier": "1.5.0",
    "typescript": "2.6.2",
    "webpack": "3.10.0",
    "webpack-dev-server": "2.9.7",
    "webpack-merge": "4.1.1"
  },
  "jest": {
    "setupFiles": [
      "./test-polyfills.js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx"
    ],
    "transform": {
      "\\.(ts|tsx)$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "/__tests__/.*\\.(ts|tsx|js|jsx)$",
    "setupTestFrameworkScriptFile": "<rootDir>/node_modules/jest-enzyme/lib/index.js",
    "moduleNameMapper": {
      "\\.(css|jpg|png)$": "<rootDir>/empty-module.js"
    },
    "snapshotSerializers": [
      "<rootDir>/node_modules/jest-serializer-enzyme"
    ],
    "unmockedModulePathPatterns": [
      "react",
      "enzyme",
      "jest-enzyme"
    ],
    "mapCoverage": true
  }
}
