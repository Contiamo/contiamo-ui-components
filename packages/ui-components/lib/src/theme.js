"use strict";
// Type definitions
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
// Default theme definition
var palette = {
    info: "#1499CE",
    success: "#00b34d",
    warning: "#FFAE00",
    error: "#DE1A1A",
    white: "#FFFFFF",
    black: "#000000",
    grey10: "#F5F5F5",
    grey20: "#F1F1F1",
    grey30: "#D0D9E5",
    grey40: "#C6D1E1",
    grey50: "#BBCADC",
    grey60: "#999999",
    grey70: "#8092B0",
    grey80: "#747474",
    grey90: "#445873"
};
var usageColors = {
    bodyText: "#555f61",
    lightText: "#969696",
    emphasizedText: "#373d3f",
    contentBorder: "#dadada",
    bodyBackground: "#F1F1F1",
    cardBackground: palette.white,
    contentSeparatorLine: "#f2f2f2",
    subContentSeparatorLine: "#f8f8f8"
};
var colors = {
    palette: palette,
    usage: usageColors
};
var baseTypography = {
    lineHeight: "1.5",
    textTransform: "none",
    letterSpacing: "normal"
};
var typography = {
    title: __assign({}, baseTypography, { fontSize: 22, fontWeight: 600 }),
    heading1: __assign({}, baseTypography, { fontSize: 13, fontWeight: 700, color: usageColors.emphasizedText }),
    heading2: __assign({}, baseTypography, { fontSize: 13, fontWeight: 600, textTransform: "uppercase", color: usageColors.lightText, "&::before": {
            content: "Â» "
        } }),
    body: __assign({}, baseTypography, { fontSize: 13, fontWeight: 400 }),
    small: __assign({}, baseTypography, { fontSize: 12, fontWeight: 400 })
};
var shadows = {
    pressed: "inset 0 1px 1px rgba(0,0,0,0.15)",
    card: "0px 1px 2px #d3d1d1"
};
var defaultTheme = {
    typography: typography,
    shadows: shadows,
    colors: colors,
    spacing: 12,
    baseZIndex: 0
};
exports.default = defaultTheme;
//# sourceMappingURL=theme.js.map