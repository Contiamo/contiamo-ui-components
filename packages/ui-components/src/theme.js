// @flow
const THEME_COLORS: COLORS = {
  primary: "#22205F",
  accent: "#3333FF",
  secondary: "#00CB98",
  tertiary: "#FFFF98",
  warn: "#DE1A1A"
}

const THEME_GREYS: GREYS = {
  "10": "#F5F5F5",
  "20": "#F1F1F1",
  "30": "#D0D9E5",
  "40": "#C6D1E1",
  "50": "#BBCADC",
  "60": "#999999",
  "70": "#8092B0",
  "80": "#747474",
  "90": "#445873",
  "100": "#2D3842",
  white: "#FFFFFF"
}

const DEFAULT_THEME: THEME = {
  colors: THEME_COLORS,
  greys: THEME_GREYS,
  fonts: {
    fontFamily: "Proxima Nova",
    fontSize: 13,
    WebkitFontSmoothing: "subpixel-antialiased",
    textRendering: "optimizeLegibility"
  },
  spacing: 12,
  baseZIndex: 0
}

export default DEFAULT_THEME
