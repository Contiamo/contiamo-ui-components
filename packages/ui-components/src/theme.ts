const palette: Palette = {
  info: "#1499CE",
  success: "#00b34d",
  warning: "#FFAE00",
  error: "#DE1A1A",
  white: "#FFFFFF",
  black: "#000000",
  grey10: "#F5F5F5",
  grey20: "#F1F1F1",
  grey30: "#D0D9E5",
  grey40: "#C6D1E1",
  grey50: "#BBCADC",
  grey60: "#999999",
  grey70: "#8092B0",
  grey80: "#747474",
  grey90: "#445873",
}

const usageColors: UsageColors = {
  bodyText: "#555f61",
  lightText: "#969696",
  emphasizedText: "#373d3f",
  contentBorder: "#dadada",
  bodyBackground: palette.grey20,
  cardBackground: palette.white,
  contentSeparatorLine: "#f2f2f2",
  subContentSeparatorLine: "#f8f8f8",
}

const colors: ThemeColors = {
  palette,
  usage: usageColors
}

const baseTypography: { lineHeight: string; textTransform: string; letterSpacing: number | "normal" } = {
  lineHeight: "1.5",
  textTransform: "none",
  letterSpacing: "normal",
}

const typography: ThemeTypography = {
  title: {
    ...baseTypography,
    fontSize: 22,
    fontWeight: 600,
  },
  heading1: {
    ...baseTypography,
    fontSize: 13,
    fontWeight: 700,
    color: usageColors.emphasizedText,
  },
  heading2: {
    ...baseTypography,
    fontSize: 13,
    fontWeight: 600,
    textTransform: "uppercase",
    color: usageColors.lightText,
    "&::before": {
      content: "Â» ",
    },
  },
  body: {
    ...baseTypography,
    fontSize: 13,
    fontWeight: 400,
  },
  small: {
    ...baseTypography,
    fontSize: 12,
    fontWeight: 400,
  },
}

const shadows = {
  pressed: "inset 0 1px 1px rgba(0,0,0,0.15)",
  card: "0px 1px 2px #d3d1d1",
}

const defaultTheme: Theme = {
  typography,
  shadows,
  colors,
  spacing: 12,
  baseZIndex: 0,
}

export default defaultTheme
