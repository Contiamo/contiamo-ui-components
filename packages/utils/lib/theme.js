"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
// Wrap each ui component in its own theme provider to make sure the default
// Contiamo theme is always available. Props are passed along unaltered.
exports.wrapTheme = function (theme) { return function (Comp) {
    return function (props) { return (React.createElement(glamorous_1.ThemeProvider, { theme: theme },
        React.createElement(Comp, __assign({}, props)))); };
}; };
//# sourceMappingURL=theme.js.map