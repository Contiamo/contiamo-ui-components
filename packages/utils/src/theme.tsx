import * as React from "react"
import { ThemeProvider } from "glamorous"

// Wrap each ui component in its own theme provider to make sure the default
// Contiamo theme is always available. Props are passed along unaltered.
export const wrapTheme = (theme: any) => (Comp: any): any => {
  return (props: any) => (
    <ThemeProvider theme={theme}>
      <Comp {...props} />
    </ThemeProvider>
  )
}
