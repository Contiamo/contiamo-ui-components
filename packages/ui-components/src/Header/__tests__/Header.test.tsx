import * as React from "react"
import { render } from "enzyme"

import wrapDefaultTheme from "../../../utils/wrap-default-theme"
import ThemelessHeader from "../Header"

import theme from "../../theme"

const Header = wrapDefaultTheme(ThemelessHeader)

describe("Header", () => {
  it("Header component renders", () => {
    const output = render(<Header className="hi">Hello</Header>)
    expect(output).toMatchSnapshot()
  })
})
