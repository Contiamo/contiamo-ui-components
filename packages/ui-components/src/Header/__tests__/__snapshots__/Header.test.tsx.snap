// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header component renders 1`] = `
<ThemeProvider
  theme={
    Object {
      "baseZIndex": 0,
      "colors": Object {
        "black": "#000000",
        "error": "#DE1A1A",
        "grey10": "#F5F5F5",
        "grey20": "#F1F1F1",
        "grey30": "#D0D9E5",
        "grey40": "#C6D1E1",
        "grey50": "#BBCADC",
        "grey60": "#999999",
        "grey70": "#8092B0",
        "grey80": "#747474",
        "grey90": "#445873",
        "info": "#1499CE",
        "success": "#689F2C",
        "warning": "#FFAE00",
        "white": "#FFFFFF",
      },
      "spacing": 12,
      "typography": Object {
        "body": Object {
          "fontSize": 13,
          "fontWeight": 400,
          "letterSpacing": 0.2,
          "lineHeight": "1.5",
          "opacity": 1,
          "textTransform": "none",
        },
        "heading1": Object {
          "fontSize": 13,
          "fontWeight": 700,
          "letterSpacing": 0.2,
          "lineHeight": "1.5",
          "opacity": 1,
          "textTransform": "none",
        },
        "heading2": Object {
          "fontSize": 13,
          "fontWeight": 700,
          "letterSpacing": 0.2,
          "lineHeight": "1.5",
          "opacity": 0.6,
          "textTransform": "none",
        },
        "small": Object {
          "fontSize": 12,
          "fontWeight": 400,
          "letterSpacing": 0.2,
          "lineHeight": "1.5",
          "opacity": 1,
          "textTransform": "none",
        },
      },
    }
  }
>
  <glamorous(Header)
    className="hi"
  >
    Hello
  </glamorous(Header)>
</ThemeProvider>
`;
